src/Common/src/System/Globalization/FormatProvider.Number.cs

```diff
@@ -1341,17 +1374,17 @@ internal static unsafe void NumberToStringFormat(ref ValueStringBuilder sb, Numb
                         }
                     }
                 }
-
+            
                 if (number.sign && section == 0)
                     sb.Append(info.NegativeSign);
 
                 bool decimalWritten = false;
 
-                fixed (char* pFormat = format)
+                fixed (char* pFormat = &format.DangerousGetPinnableReference())
                 {
                     char* cur = dig;
```

Comment:
Are `cur`/`dig` subject to the same "there isn't necessarily a `\0`"?  Can/should they be switched to be Span instead of `char*`?

Reply:
> Are cur/dig subject to the same "there isn't necessarily a \0"

No, they will be null-terminated:
https://github.com/stephentoub/corefx/blob/e9edba407e4b0119bf09c5cf0f8f44dc3509228e/src/System.Runtime.Numerics/src/System/Numerics/BigNumber.cs#L695-L703
But it'd be great subsequently to revisit this code and move away from all of this unsafeness.  It's very convoluted.